Math/NumberTheory/Prefactored.hs

```diff
@@ -100,15 +100,15 @@ fromValue a = Prefactored a [(a, 1)]
 -- > > fromFactors (splitIntoCoprimes [(140, 2), (165, 3)])
 -- > Prefactored {prefValue = 88045650000, prefFactors = [(5, 5), (28, 2), (33, 3)]}
 fromFactors :: Integral a => [(a, Word)] -> Prefactored a
-fromFactors as = Prefactored (product (map (uncurry (^)) as)) (splitIntoCoprimes as)
+fromFactors as = Prefactored (product (map (uncurry (^)) as)) (toList $ splitIntoCoprimes as)
```

Comment:
Actually since `prefFactors` keeps "List of pairwise coprime (but not neccesarily prime) factors, accompanied by their multiplicities", we should change its type from `[(a, Word)]` to `Coprimes a Word`. In such case user can analyse contents of `Prefactored` and be sure that it contains what it declares to contain.

And other way around: let `fromFactors` accept `Coprimes a Word`. User should be responsible to provide it with a correct input.